cypress/backend/pynn/pynn.cpp

```diff
@@ -324,6 +326,43 @@ std::unordered_set<const NeuronType *> PyNN::supported_neuron_types() const
 	return SUPPORTED_NEURON_TYPE_MAP.find("__default__")->second;
 }
 
+namespace {
+/**
+ * Helper function to put writing of network description into a seperate thread
+ */
+void pipe_write_helper(std::string file, NetworkBase &source)
+{
+	std::filebuf fb_in;
+	fb_in.open(file, std::ios::out);
+	std::ostream data_in(&fb_in);
+	if (data_in.good() == false) {
+		throw ExecutionError("Error in pipe to python");
+	}
+	// Send the network description to the simulator, inject the connection
+	// transformation to rewrite the connections
+	binnf::marshall_network(source, data_in);
+	fb_in.close();
+}
+
+std::map<std::string, std::string> fifo_filenames(const std::string log_path)
```

Comment:
Why is this parameter called `log_path`? Just call it `path`. There is no semantic relationship between the fifo and the log at this point.